### PR TITLE
Migrate off deprecated DefaultProgramRunner

### PR DESCRIPTION
**Pull request by abrooksv**
Migrated off deprecated method.

Also switched it to async so that invoking the state is on BG thread always

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/remote/RemoteLambdaRunner.kt [hunk 1]
```diff
@@ -4,14 +4,39 @@
...
+        DefaultRunExecutor.EXECUTOR_ID == executorId && profile is RemoteLambdaRunConfiguration
+
+    override fun execute(environment: ExecutionEnvironment, state: RunProfileState): Promise<RunContentDescriptor?> {
+        val runPromise = AsyncPromise<RunContentDescriptor?>()
+        ApplicationManager.getApplication().executeOnPooledThread {
+            val builder = state.execute(environment.executor, this)?.let {
+                RunContentBuilder(it, environment)
+            }
+
+            builder?.let {
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/remote/RemoteLambdaRunner.kt:33]: does the promise needs to be failed if the builder is null?

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/remote/RemoteLambdaRunner.kt]: Our code can't return null, we can probably downcast it to make it safe

**Review by rli — APPROVED** [review]

**Review by rli — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/remote/RemoteLambdaRunner.kt (hunk 1 above)
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/remote/RemoteLambdaRunner.kt]: No longer possible if we downcast, and thats guarded by the canRun method so we good :D

**Review by rli — APPROVED** [review]

**Pull request merged**